contracts/TalentLayerEscrow.sol

```diff
@@ -896,12 +925,18 @@ contract TalentLayerEscrow is
      */
     function _reimburse(uint256 _transactionId, uint256 _amount) private {
         Transaction storage transaction = transactions[_transactionId];
-        uint256 totalReleaseAmount = _calculateTotalWithFees(
+        uint256 reimbursedReferralAmount;
+
+        if (transaction.referrerId != 0) {
```

Comment:
- move calculate part inside _calculateTotalWithFees
- don't pass all values inside the function, just the _amount, and call directly Transaction storage transaction = transactions[_transactionId]; from it